Fix ending number as time

iamnwi opened the pull request:
fix case "move 3pm appointment to 4"

Commit:
fix 'move 3pm appointment to 4'

Commit:
Merge branch 'ms_master' into fix_ending_number_as_time

Commit:
refined a regex

Commit:
Merge branch 'ms_master' into fix_ending_number_as_time

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs:
```diff
@@ -97,6 +98,10 @@ public class EnglishTimeExtractorConfiguration : ITimeExtractorConfiguration
 
+        public static readonly Regex SpecialTimePattern =
+            new Regex(DateTimeDefinitions.SpecialTimePattern,
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Rename it to NumberEndingPattern, the same to all SpecialTimePattern.
This is not a good naming.

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs:
```diff
@@ -140,10 +145,16 @@ public class EnglishTimeExtractorConfiguration : ITimeExtractorConfiguration
 
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
+
+        public IExtractor NumExtractor => NumberExtractor.GetInstance();
```
These should be in the construction method. Please follow the initialization of the durationExtractor in this file.

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs:
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
             }
+            // handle special patterns
+            //Regex r = new Regex($@"\b((?<oldTime>.*)\s+(?<meeting>meeting|appointment|conference|assembly)\s+to\s+(?<newTime>.*)(.|,)?)\b",
+            //RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Please delete the commented code.

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs:
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
+                // check if oldTime string is a time format
+                var oldTime = match.Groups["oldTime"];
+                var timeRes = this.config.TimeExtractor.Extract(oldTime.ToString());
+                if (timeRes.Count == 0) continue;
```
if (timeRes.Count == 0) 
{
continue;
}

always using {} after "if"

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs:
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
+                // check if newTime string is a number format
+                var newTime = match.Groups["newTime"];
+                var numRes = this.config.NumExtractor.Extract(newTime.ToString());
+                if (numRes.Count == 0) continue;
```
same as above

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs:
```diff
@@ -0,0 +1,12 @@
+﻿// Enable a reference time pass to the extractor
```
This should be deleted since we do the IDatetimeExtractor in another PR.

iamnwi replied:
I didn't see it in my project, maybe it still exist in folder, do I need to delete it in folder also?

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/French/Extractors/FrenchTimeExtractorConfiguration.cs:
```diff
@@ -172,9 +177,16 @@ public class FrenchTimeExtractorConfiguration : ITimeExtractorConfiguration
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
+
+        public IExtractor NumExtractor => new NumberExtractor();
+
```
same as above

rubio41 commented on .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishTimeExtractorConfiguration.cs:
```diff
@@ -89,10 +92,16 @@ public class SpanishTimeExtractorConfiguration : ITimeExtractorConfiguration
 
         Regex ITimeExtractorConfiguration.IshRegex => null;
 
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
```
same as above

Commit:
refined code

tellarin committed:
Fixing 2 broken unit tests in French (#166)

* - Fixing broken currency test;
- Workaround for failing test in numbers (IPs shouldn't be recognized as numbers, or all 4 segments should be recog separately); temporarily reverting to same behaviour as other languages.

* Ignoring in JS temporarily.

Commit:
Merge remote-tracking branch 'upstream/master' into fix_ending_number_as_time

Commit:
build the yaml file

Commit:
move the number ending function to the mergedextractor due to it relied on the time extractor

Commit:
style fixing

rubio41 committed:
Fix ending number as time (#168)

* fix 'move 3pm appointment to 4'

* refined a regex

* refined code

* build the yaml file

* move the number ending function to the mergedextractor due to it relied on the time extractor

* style fixing

pcostantini committed:
CI - .NET Test Fails should invalidate build (#169)

* CI Fix to report failure when a .NET test fails

* Run .NET tests in parallel

Commit:
Update Italian-NumbersWithUnit.yaml (#170)

Commit:
modified one regex and added some new test cases accordingly

Commit:
added a parsing function for this case

Commit:
merged from the latest MS master version
